src/devices/Mcp3008/Mcp3008.cs

```diff
@@ -29,41 +23,36 @@ public void Dispose()
             _spiDevice = null;
         }
 
-        public int Read(int channel, InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
+        public int Read(int channel)
         {
             if (channel < 0 || channel > 7)
             {
                 throw new ArgumentException("ADC channel must be within 0-7 range.");
             }
 
-            return ReadSpi(channel, inputConfiguration);
-        }
+            Span<byte> writeBuffer = stackalloc byte[3];
```

Comment:
question: stackalloc would initialize all bytes to 0 right?

Reply:
yes, it must, otherwise you could leak some sensitive data by accident and that would be security issue